README.md

```diff
@@ -72,3 +72,19 @@ render((
   </Router>
 ), document.getElementById('root'))
 ```
+
+#### If you're using React Router 4's BrowserRouter you can incorporate qhistory like this:
+
+```js
+class QueryRouter extends BrowserRouter {
```

Comment:
The `<BrowserRouter>` is just a thin wrapper around the `<Router>` class. If you want to have a `<QueryRouter>` component, I would recommend just copying the code from [`<BrowserRouter>`](https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/BrowserRouter.js), but modifying the history creation line.

This example is basically creating a browser history (when `super` is called), then overwriting it. Copying the code from `BrowserRouter.js` isn't much longer and avoids creating an extra `history` object.

Reply:
Yeah, I considered that approach but it felt very wrong to have to ignore the Router they've designed for use in a browser just to be able to do something as basic as reading query params as a plain JavaScript object.

Have you seen any better/cleaner approaches to this issue?

Reply:
In general, the HOC approach does make sense to me... I guess it just *feels* wrong to make a parallel HOC to `BrowserRouter` but perhaps that's just my sensibilities from non-component-oriented design speaking.

I'll take a second pass at this. Thanks for the feedback.

Reply:
The `<BrowserRouter>` is just a convenience component, but it really isn't anything special.

To be perfectly honest, I'm not sure why you would want to use a `<QueryRouter>` when you can just use `<Router history={history}>`. This pattern is fairly commonly used (that _is_ what a `<BrowserRouter>` renders). Any Redux users who want to access the `history` object outside of the store are advised to do this.

Perhaps instead of proposing a `<QueryRouter>` component, it might be better to just update the README to specify that `qhistory` should only be used with a `<Router>` and not one of the specialty `<___Router>` components?